libcontainer/cgroups/systemd/common.go

```diff
@@ -293,8 +293,14 @@ func generateDeviceProperties(r *configs.Resources) ([]systemdDbus.Property, err
 			// rules separately to systemd) we can safely skip entries that don't
 			// have a corresponding path.
 			if _, err := os.Stat(entry.Path); err != nil {
-				logrus.Debugf("skipping device %s for systemd: %s", entry.Path, err)
-				continue
+				// Also check /sys/dev so that we don't depend on /dev/{block,char} being populated.
+				// (/dev/{block,char} is populated by udev, which isn't strictly required for systemd).
+				// Ironically, this happens most easily when starting containerd within a runc created
+				// container itself.
+				if _, err := os.Stat("/sys" + entry.Path); err != nil {
```

Comment:
Please use securejoin

https://github.com/opencontainers/runc/search?q=securejoin

Reply:
Done. Wasn't sure we needed to do that since entry.Path is created in the lines right above this and thusly we know it's content is safe.

Reply:
> Done. Wasn't sure we needed to do that since entry.Path is created in the lines right above this and thusly we know it's content is safe.

I concur.